packages/grafana-ui/src/components/uPlot/plugins/CursorPlugin.tsx

```diff
@@ -0,0 +1,112 @@
+import React, { useState, useEffect, useRef, useCallback } from 'react';
+
+import { PlotPluginProps } from '../types';
+import { pluginLog } from '../utils';
+import { usePlotPluginContext } from '../context';
+
+interface CursorPluginAPI {
+  focusedSeriesIdx: number | null;
+  focusedPointIdx: number | null;
+
+  coords: {
+    // coords relative to plot canvas, css px
+    plotCanvas: Coords;
+    // coords relative to viewport , css px
+    viewport: Coords;
+  };
+}
+
+interface CursorPluginProps extends PlotPluginProps {
+  onMouseMove?: () => void; // anything?
+  children: (api: CursorPluginAPI) => React.ReactElement | null;
+  // on what interaction position should be captures
+  capture?: 'mousemove' | 'mousedown';
+  // should the position be persisted when user leaves canvas area
+  lock?: boolean;
+}
+
+interface Coords {
+  x: number;
+  y: number;
+}
+
+// Exposes API for Graph cursor position
+export const CursorPlugin: React.FC<CursorPluginProps> = ({ id, children, capture = 'mousemove', lock = false }) => {
+  const pluginId = `CursorPlugin:${id}`;
+  const plotCanvas = useRef<HTMLDivElement>(null);
+  const plotCanvasBBox = useRef<any>({ left: 0, top: 0, right: 0, bottom: 0, width: 0, height: 0 });
+  const pluginsApi = usePlotPluginContext();
+
+  // state exposed to the consumers, maybe better implement as CursorPlugin?
+  const [focusedSeriesIdx, setFocusedSeriesIdx] = useState<number | null>(null);
+  const [focusedPointIdx, setFocusedPointIdx] = useState<number | null>(null);
+  const [coords, setCoords] = useState<{ viewport: Coords; plotCanvas: Coords } | null>(null);
```

Comment:
wonder if it's not better to have a single state object here, react cannot batch state updates to useState updates unless they happen in a react synthetic event. So need to watch out for double/triple renders when using separate useState for every property.